### ISSUE TITLE
CodePush.getCurrentPackage() not found.

### ISSUE THREAD

**Issue by livioso** [Microsoft/react-native-code-push#195]
Hello everybody. 

I am encountering the following problem when I run my application on the device *(simulator works fine)*.  I am using **React Native 0.19** and **react-native-code-push 1.7.2-beta**. Not sure but I think I had the same error before *(with React Native 0.18 and react-native-code-push 1.6.0-beta).*

```
2016-02-11 16:37:40.259 Fashwell[778:256664] Loading JS bundle from file:///var/mobile/Containers/Bundle/Application/C840798E-87AA-4700-BC2A-1A7A89453BC7/Fashwell.app/main.jsbundle
2016-02-11 16:37:40.676 [warn][tid:com.facebook.React.JavaScript] Callback with id 14: CodePush.getCurrentPackage() not found
2016-02-11 16:37:40.683 [error][tid:com.facebook.React.RCTExceptionsManagerQueue] Unhandled JS Exception: Callback with id 14: CodePush.getCurrentPackage() not found
2016-02-11 16:37:40.683 Fashwell[778:256712] *** Terminating app due to uncaught exception 'RCTFatalException: Unhandled JS Exception: Callback with id 14: CodePush.getCurrentPackage() not found', reason: 'Unhandled JS Exception: Callback with id 14: CodePush.getCurrentPackage() n..., stack:
value@1:28822
<unknown>@1:26937
_@1:25866
value@1:26907
'
*** First throw call stack:
(0x183999900 0x183007f80 0x183999848 0x10010902c 0x100103a1c 0x18399f430 0x18389ceb4 0x1838a17e0 0x100107180 0x100127010 0x100126a00 0x1003f1bf0 0x1003f1bb0 0x1003fe6c8 0x1003f58a0 0x1003f1bb0 0x100400e10 0x1004004d8 0x183601470 0x183601020)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb) 
```

In my main component's `componentWillMount()` I simply do the following:
```
codePush.sync();
```

#### Update

1. Apparently, I get the same error in the simulator (after / while closing the app).

2. When compiled with Debug I get a slightly more verbose output:

```
2016-02-11 17:13:24.235 [trace][tid:com.facebook.React.JavaScript] [CodePush] Checking for update.
2016-02-11 17:13:24.261 [trace][tid:com.facebook.React.JavaScript] [CodePush] An unknown error occurred.
2016-02-11 17:13:24.262 [trace][tid:com.facebook.React.JavaScript] [CodePush] The file “app.json” couldn’t be opened because there is no such file.
2016-02-11 17:13:24.262 [warn][tid:com.facebook.React.JavaScript] Callback with id 18: CodePush.getCurrentPackage() not found
2016-02-11 17:13:24.301 [error][tid:com.facebook.React.RCTExceptionsManagerQueue] Unhandled JS Exception: Callback with id 18: CodePush.getCurrentPackage() not found
2016-02-11 17:13:24.463 [info][tid:com.facebook.React.JavaScript] Unable to load source map: No source map URL found. May be running from bundled file.
```

I really don't have a clue how to debug this. Help / inputs would be highly appreciated.

Thanks,
Livio

**Comment by lostintangent** [comment]
Hey @livioso! We haven't seen this error before, so I'm looking into it right now. Just to confirm, do you see this error after you start the app and `sync` is called? You don't need to do anything else to repro it?

Also, one other question, did you previously install any other CodePush updates in this app?

**Comment by geof90** [comment]
Hi @livioso,

Check out this SO post as well as this blog.
http://stackoverflow.com/questions/34223850/react-native-invariant-violation-callback-with-id-2-error
http://jairtrejo.mx/blog/2015/09/console-error-callback-not-found

Do you perhaps have 2 debugger sessions running?

**Comment by lostintangent** [comment]
Hey @livioso! Thank you so much for the detail in this. We were able to identify the cause of the "Callback not found error" thanks to this and we will be submitting a PR for that ASAP.

The issue about the missing "app.json" file, however, is very interesting. Would you be able to capture an [xcappdata](https://developer.apple.com/library/ios/recipes/xcode_help-devices_organizer/articles/manage_containers.html) file of your app via the Xcode Devices window and send that to me at joncart@microsoft.com? That would be really useful for diagnosing how the app might have got into a bad state.

Thank you so much!

**Comment by livioso** [comment]
Hey @lostintangent.

Thanks a lot for your fast reply. 

I'm unable to reproduce the error this morning after a `rm -rf ./node_modules/`, `npm install`, `Product > Clean` in Xcode and `rm -rf ~/Library/Developer/XCode/DerivedData/`. 

The error is magically gone... 🤔

**Comment by lostintangent** [comment]
OK cool. That makes sense because your error indicates that the app somehow got into a bad state, and doing a clean across the board would resolve that. That said, PR #196 resolves the crash that will happen when the app gets into this state and we're working to figure out how he "app.json" file went missing.

Thanks again so much for reporting this!

**Comment by lostintangent** [comment]
This error has been addressed with #196 and will be released to NPM soon. Thanks for the feedback and let us know if you hit this problem (or any other problem) again.

**Comment by livioso** [comment]
Alright. Thanks again for the fast support!
